src/linux/cgroups2.hpp

```diff
@@ -217,9 +217,49 @@ Try<BandwidthLimit> max(const std::string& cgroup);
 
 namespace memory {
 
+// Memory usage limit.
+//
+// Represents a snapshot of "memory.high" and "memory.max".
+struct ByteLimit
+{
+  // Constructs a limitless byte limit.
+  ByteLimit() = default;
+
+  ByteLimit(const Bytes& limit);
+
+  bool operator==(const ByteLimit& other) const;
+
+  // Limit in bytes. None if the limit is "unlimited".
+  Option<Bytes> limit;
+};
```

Comment:
what's up with this? is this only to avoid `Try<Option<Bytes>>`? If so, that's what `Result<Bytes>` provides:

```
Try<Nothing> set_max(const std::string& cgroup, const Option<Bytes>& limit);
Result<ByteLimit> max(const std::string& cgroup);
```

Reply:
Similar to `cpu::BandwidthLimit`, a `ByteLimit` can either be some byte amount or "max". The idea is to be more explicit about Option<Bytes> representing an uncapped limit. 

IIUC we would then have:
```c++
Try<Nothing> set_max(const std::string& cgroup, const Option<Bytes>& limit);
Result<Bytes> max(const std::string& cgroup); // instead of Result<ByteLimit>? 
```

Reply:
yeah, s/ByteLimit/Bytes/ in my snippet, whoops